sklearn/utils/estimator_checks.py

```diff
@@ -1630,6 +1631,51 @@ def transform(self, X):
                     shallow_params.items()))
 
 
+def check_set_params(name, Estimator):
+    # Trivial check to make sure set_params is working
+    classifier = LinearDiscriminantAnalysis()
+    with ignore_warnings(category=DeprecationWarning):
+        if name in META_ESTIMATORS:
+            estimator = Estimator(classifier)
+        else:
+            estimator = Estimator()
+
+    params = estimator.get_params()
+    estimator.set_params(**params)
+
+    # Check that get_params() returns the same thing
+    # before and after set_params()
+
+    test_values = [-np.inf, np.inf, None,
+                   -100, 100, -0.5, 0.5, 0,
+                   "", "value",
+                   ('a', 'b'), {'key': 'value'}]
+
+    for param_name in params.keys():
+        for value in test_values:
+
+            try:
+                estimator.set_params(**{param_name: value})
+                get_value = estimator.get_params()[param_name]
+            except:
+                # triggered some parameter validation
```

Comment:
Sorry, I didn't realize there had been comments.

Here we don't error because we just want to check that `get_value` is `value` if `value` is successfully set.